### PR TITLE
Add `replace_lead()` to content working group

### PR DESCRIPTION
**Pull request by mnaamani**
Implements #243

### CODE REVIEW

File: runtime-modules/content-working-group/src/lib.rs [hunk 1]
```diff
@@ -2122,6 +2042,98 @@ impl<T: Trait> versioned_store_permissions::CredentialChecker<T> for Module<T> {
...
+
+        // Store lead
+        <LeadById<T>>::insert(new_lead_id, new_lead);
+
+        // Update current lead
+        <CurrentLeadId<T>>::put(new_lead_id); // Some(new_lead_id)
+
+        // Update next lead counter
+        <NextLeadId<T>>::mutate(|id| *id += <LeadId<T> as One>::one());
+
```
Comment (shamil-gadelshin) [inline runtime-modules/content-working-group/src/lib.rs]: Probably, assert() can be replaced with "?" function invocation. Code reordering is required.

File: runtime-modules/content-working-group/src/lib.rs [hunk 2]
```diff
@@ -2122,6 +2042,98 @@ impl<T: Trait> versioned_store_permissions::CredentialChecker<T> for Module<T> {
...
+        // Unregister from role in membership model
+        let current_lead_role = role_types::ActorInRole {
+            role: role_types::Role::CuratorLead,
+            actor_id: lead_id,
+        };
+
+        let unregistered_role = <members::Module<T>>::unregister_role(current_lead_role).is_ok();
+
+        assert!(unregistered_role);
```
Comment (shamil-gadelshin) [inline runtime-modules/content-working-group/src/lib.rs]: Probably, assert() can be replaced with "?" function invocation. Code reordering is required.

File: runtime-modules/content-working-group/src/lib.rs (hunk 1 above)
Comment (mnaamani) [reply runtime-modules/content-working-group/src/lib.rs]: Thats a good point, I think this is the new convention we decided on. However here I avoided changing the implementation of set_lead and unset_lead I just relocated them and converted them from being dispatchables to be private methods on the `Module`.

I can update it not much extra work.

Comment (mnaamani) [reply runtime-modules/content-working-group/src/lib.rs]: done in e4f57a28617fb9b66734164f9707ad5dd110e4be

File: runtime-modules/content-working-group/src/lib.rs (hunk 2 above)
Comment (mnaamani) [reply runtime-modules/content-working-group/src/lib.rs]: done in e4f57a28617fb9b66734164f9707ad5dd110e4be

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**